### PR TITLE
feat(aws-auth-cognito): Adds clientMetadata to AWSCognitoAuthSignUpOptions

### PR DESCRIPTION
**Pull request by haverchuck**
*Description of changes:*
Adds clientMetadata to AWSCognitoAuthSignUpOptions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/AuthComponentTest.java [hunk 1]
```diff
@@ -196,15 +196,16 @@ public void signUp() throws AuthException {
...
             callback.onResult(amcResult);
             return null;
         }).when(mobileClient)
-            .signUp(any(), any(), any(), any(), Mockito.<Callback<SignUpResult>>any());
+            .signUp(any(), any(), any(), any(), any(), Mockito.<Callback<SignUpResult>>any());
 
         AWSCognitoAuthSignUpOptions options = AWSCognitoAuthSignUpOptions.builder()
                 .userAttribute(AuthUserAttributeKey.email(), ATTRIBUTE_VAL)
                 .validationData(METADATA)
+                .clientMetadata(METADATA)
```
Comment (rjuliano) [inline aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/AuthComponentTest.java:208]: Maybe use a different value for clientMetadata just to be on the safe side?

**Review by rjuliano — APPROVED** [review]
One minor comment, but other than that LGTM. 🚢

**Review by richardmcclellan — APPROVED** [review]
Thanks for adding this!

**Comment by haverchuck** [comment]
@rjuliano I responded to your PR feedback, please take a look at your convenience!

**Comment by richardmcclellan** [comment]
@haverchuck the unit tests failed due to a checkstyle issue:
```
> Task :aws-auth-cognito:checkstyle
--
461 | [ant:checkstyle] [ERROR] /codebuild/output/src601626422/src/github.com/aws-amplify/amplify-android/aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/AuthComponentTest.java:138: Line is longer than 120 characters (found 124). [LineLength]
462 | [ant:checkstyle] [ERROR] /codebuild/output/src601626422/src/github.com/aws-amplify/amplify-android/aws-auth-cognito/src/test/java/com/amplifyframework/auth/cognito/AuthComponentTest.java:139: Line is longer than 120 characters (found 128). [LineLength]
```

**Pull request merged**